syntax = "proto3";

import "chromeperf/pinpoint/change.proto";
import "chromeperf/engine/task.proto";

package chromeperf.pinpoint;

// FIXME: Depend on the buildbucket proto schema?
message BuildBucketBuild {
  string id = 1;
  string url = 2;
  string status = 3;
  string result = 4;
  string result_details_json = 5;
}

message FindIsolateTaskPayload {
  chromeperf.pinpoint.Change change = 1;
  chromeperf.pinpoint.GerritPatch patch = 2;
  string builder = 3;
  string target = 4;
  string isolate_server = 5;
  string isolate_hash = 6;
  string bucket = 7;
  int32 tries = 8;
  BuildBucketBuild buildbucket_build = 9;
  repeated chromeperf.engine.ErrorMessage errors = 10;
}

message BuildUpdate {
  // The build ID.
  string id = 1;

  // A state associated with this build-related update.
  // TODO: Sync this with what we'll get from the BuildBucket Pub/Sub messages.
  enum BuildState {
    UNSPECIFIED_BUILDSTATE = 0;
    PENDING = 1;
    RUNNING = 2;
    FAILED = 3;
    COMPLETED = 4;
    CANCELLED = 5;
  }
  BuildState state = 2;
}


// A subset of FindIsolateTaskPayload.
// TODO: refactor FindIsolateTaskPayload to have an Input submessage, and delete
// this?
message FindIsolateTaskOptions {
  string builder = 1;
  string target = 2;
  string bucket = 3;
  chromeperf.pinpoint.Change change = 4;
}
